Sync `html/select` from WPT Upstream

Ahmad-S792 opened the pull request:
#### 54017ed304f22ac7519b09fd9371eb86451e3a4e
<pre>
Sync `html/select` from WPT Upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=301047">https://bugs.webkit.org/show_bug.cgi?id=301047</a>
<a href="https://rdar.apple.com/162943410">rdar://162943410</a>

Reviewed by Anne van Kesteren.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/19250d046daa69748ece904fb05c768713e045aa">https://github.com/web-platform-tests/wpt/commit/19250d046daa69748ece904fb05c768713e045aa</a>

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/html/select/select-capitalize-sizing-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/select/select-capitalize-sizing-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/select/select-capitalize-sizing.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/select/w3c-import.log: Added.
* LayoutTests/tests-options.json:

Canonical link: <a href="https://commits.webkit.org/301843@main">https://commits.webkit.org/301843@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f2ec3cd99693f706f9cc7d4208dd2014359e86f2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127023 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46659 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37684 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/134026 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78587 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/9473afb3-b0a8-4aa6-8810-e02dd130993d) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47276 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55186 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96655 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/64665 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/df505740-50b6-4a1d-9991-e179a7f3ce9b) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129971 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37833 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113649 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77165 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/54d403c0-b0db-4cc9-8369-a5c8a666a18b) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36697 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/31794 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77418 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107683 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/32132 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136552 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53678 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41335 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105170 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54182 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110007 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104861 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26774 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50383 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28749 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/51196 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53610 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52849 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56182 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54607 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (approved):
I don't really understand why we have html/select and html/semantics/forms/the-select-element. These should prolly be moved into the latter directory?

Ahmad-S792 committed:
Sync `html/select` from WPT Upstream
https://bugs.webkit.org/show_bug.cgi?id=301047
rdar://162943410

Reviewed by Anne van Kesteren.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/19250d046daa69748ece904fb05c768713e045aa

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/html/select/select-capitalize-sizing-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/select/select-capitalize-sizing-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/select/select-capitalize-sizing.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/select/w3c-import.log: Added.
* LayoutTests/tests-options.json:

Canonical link: https://commits.webkit.org/301843@main
